### ISSUE TITLE
When running plot() function in a Jupyter Notebook, a greyed out window appears.

### ISSUE THREAD

**Issue by downiec** [CDAT/vcs#383]
If vcs.init() is run in a Jupyter notebook, it should automatically detect and have bg=true to avoid the greyed out window when sidecar is used in a Jupyter Notebook.

**Comment by doutriaux1** [comment]
no default needs to be bg=False as now. Look for the bit of code I added to detect notebook, for sidecar

**Comment by doutriaux1** [comment]
Actually the code should already be in place (I thought I did code this): https://github.com/CDAT/vcs/blob/master/vcs/Canvas.py#L931-L935

I wo der why it's not picking it up

**Comment by doutriaux1** [comment]
@scottwittenburg is it related to the new vtk-cdat ?

**Comment by doutriaux1** [comment]
I remember seeing a window even with mesalib, maybe it's a similar issue?

**Comment by scottwittenburg** [comment]
I think I'll need some instructions on how to reproduce this if I'm to be of any help.

**Comment by doutriaux1** [comment]
according to @downiec when in a jupyternotebook with the non-mesa env (py3) `x=vcs.init();x.plot([1,2,3])` make a window appear

**Comment by doutriaux1** [comment]
we do not want the window appear, hence bg=True

**Comment by downiec** [comment]
Could it be because my conda environment? I just remembered I'm using the old one because that's the one that's currently working.

**Comment by scottwittenburg** [comment]
Sounds good, thanks @downiec.

**Comment by downiec** [comment]
What vcs version do I need? I tried pulling from master and then running python setup.py install, but now I'm getting an error and it won't plot anything.
AttributeError: module 'vtkmodules.all' has no attribute 'vtkRenderingOpenGLPython'

**Comment by downiec** [comment]
One last comment. In the JupyterLab Notebook, first time I run the cell, I get:
`AttributeError: module 'vtkmodules.all' has no attribute 'vtkRenderingOpenGLPython'`
Second time  I run the cell (same exact code, I just run the cell again):
`ZeroDivisionError: float division by zero`
Third time I run it, it plots without an error.

**Comment by downiec** [comment]
@scottwittenburg Thanks for the response, I went ahead and removed the vcs egg files like you said, and then I ran python setup.py install. That all fixed the error, thanks! Unfortunately it doesn't resolve the greyed out window issue. Basically, if I run the following in a Jupyter Lab Notebook:
canvas.plot(clt,"default","default")
The plot is made, no problem. But then there's also a separate greyed out window that pops up and it can't be closed or minimized etc, it's just there. I tried running:
canvas.plot(clt,"default","default",bg=True)
And the issue still persists, with a greyed out window popping up. The window only goes away when I restart the kernel.

**Comment by scottwittenburg** [comment]
@downiec Can you run the following in the same environment you just cleaned up, and see if the window pops up?

```python
import vcs
canvas = vcs.init(bg=1)
canvas.plot([1, 2, 3, 4, 5])
canvas.png("simple.png")
```

Let me know what happens, thanks.

**Comment by downiec** [comment]
Yeah the window pops up both in a JupyterLab environment and also in the console.
